cfy_manager/components/restservice/db.py

```diff
@@ -214,14 +255,36 @@ def _run_script(script_name, args_dict=None, configs=None):
         cmd.append(args_json_path)
 
     result = common.sudo(cmd, env=env_dict)
-
-    _log_results(result)
+    return _extract_output_and_log_results(result, extract_output_func)
 
 
 def populate_db(configs):
+    def get_reporter_and_token(string):
+        token_len = UUID4HEX_LEN + ENCODED_USER_ID_LENGTH
+        _reporter = string[:- token_len - 1]
+        _token = string[- token_len:]
+        return _reporter, _token
+
+    def is_reporter_token(line):
+        return get_reporter_and_token(line)[0] in reporter_to_conf_key
+
+    reporter_to_conf_key = {
+        'db_status_reporter_token': DB_STATUS_REPORTER,
+        'queue_status_reporter_token': QUEUE_STATUS_REPORTER,
+        'manager_status_reporter_token': MANAGER_STATUS_REPORTER
+    }
+
     logger.notice('Populating DB and creating AMQP resources...')
-    args_dict = _create_args_dict()
-    _run_script('create_tables_and_add_defaults.py', args_dict, configs)
+    args_dict = _create_populate_db_args_dict()
+    tokens = _run_script('create_tables_and_add_defaults.py',
```

Comment:
i don't like this idea of parsing the output at all; if we need to get the output from that script, i say make the script return json, which is going to be a more structured way than examining stdout log lines

to do that, make the logging go to stderr instead (by using logging instead of print in the script, and something like `logging.basicConfig(stream=sys.stderr, level=logging.INFO)`)

this is a good idea in general, to make the script work in terms of "data to stdout, logs to stderr" fashion

if we do this, then the `is_..` function from here, and the "extract predicate func" will not be necessary

Reply:
I'll refactor this